### ISSUE TITLE
server_name variable not properly replaced with correct value in chef-server/omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx.conf.erb template, making nginx fail with "nginx: [emerg] directive "server_name" is not terminated by ";" in /var/opt/opscode/nginx/etc/nginx.conf" error.

### ISSUE THREAD

**Issue by adolfoasp** [chef/chef-server#1407]
## Expected Behavior

In title metioned template, for example in that block:

    # We support three options: serve nothing on non_ssl_port (80),
    # redirect to https, or actually serve the API.
    <%- if @non_ssl_port -%>
      <%- if @enable_non_ssl -%>
        include <%= @chef_http_config %>;
      <%- else -%>
          server {
            <%= @helper.listen_port('http') %>
            server_name <%= @helper.server_name %>;
            access_log /var/log/opscode/nginx/rewrite-port-80.log;
            return 301 https://$host$request_uri;
          }
      <%- end -%>
    <%-  end %>

After executing chef-server-ctl reconfigure, we should obtain in <%= @helper.server_name %>; the value of nginx['server_name'] variable that we have set in /etc/opscode/chef-server.rb variable with no more values. 

## Current Behavior

Currently, we have set the variable in /etc/opscode/chef-server.rb to:
nginx['server_name'] = "chef.mycustomdomain.com"
After chef-server-ctl reconfigure, we got the final value of:
```
server_name chef.mycustomdomain.com localhost 127.0.0.1 IP_ADRESS {"ip_address"=>"IP_ADRESS", "type"=>"public", "netmask"=>"IP_ADRESS_NETMASK"}
{"ip_address"=>"IPV6_ADRESS", "type"=>"private", "cidr"=>CIRDVALUE};
```

Where IP_ADRESS, IP_ADRESS_NETMASK, IPV6_ADRESS, and CIRDVALUE are real values related to our server.

This behaviour is related to the usage of implicit_hosts definition in chef-server/omnibus/files/private-chef-cookbooks/private-chef/libraries/nginx_erb.rb library. We suspect that implicit_hosts definition is taking more attributes than it should take.

## Steps to Reproduce (for bugs)

1. This is happening in a Digital Ocean Dropler, with normal chef-server.rb config and chef-server installed by apt source list.

## Your Environment

* Chef Server Version: 12.17.3-1
* Total/free RAM and disk space: Not-Important
* Operating System and Version: Ubuntu 14.04 LTS
* If upgrading, previous Chef Server version: 12.16.9-1 
* Running in a container? No.

**Comment by stevendanna** [comment]
Apologies for this.  We are shipping a change to the default config to address this shortly.

You can fix this by setting:

```
nginx['use_implicit_hosts'] = false
```

in chef-server.rb and running reconfigure.

Unfortunately we added a feature that collects information about the cloud that you are running on to generate nginx configuration; however, the digital ocean cloud plugin doesn't provide data in a format that the code expects.

**Comment by stevendanna** [comment]
We'll be setting use_implicit_hosts to false until we can ensure it works on digital ocean: 

https://github.com/chef/chef-server/pull/1408

**Comment by belen0** [comment]
After setting to false that attribute, now our hosted server is working properly.

Thanks

**Comment by adolfoasp** [comment]
Thanks a lot! Setting to false the attribute, the template uses the correct server_name variable.

**Comment by stevendanna** [comment]
We fixed this in 12.17.5 by setting the default to `false`.
